src/appmixer/front/lib.js

```diff
@@ -0,0 +1,152 @@
+        } else {
+            throw new context.CancelError('Unsupported outputType ' + outputType);
+        }
+    },
+
+    getProperty(obj, path) {
+        return path.split('.').reduce((acc, part) => acc?.[part], obj);
+    },
+
+    getOutputPortOptions(context, outputType, itemSchema, { label }) {
+
+        if (outputType === 'object' || outputType === 'first') {
+            const options = Object.keys(itemSchema)
+                .reduce((res, field) => {
+                    const schema = itemSchema[field];
+                    const { title: label, ...schemaWithoutTitle } = schema;
+
+                    res.push({
+                        label, value: field, schema: schemaWithoutTitle
+                    });
+                    return res;
+                }, [{
+                    label: 'Current Item Index',
+                    value: 'index',
+                    schema: { type: 'integer' }
+                }, {
+                    label: 'Items Count',
+                    value: 'count',
+                    schema: { type: 'integer' }
+                }]);
+
+            return context.sendJson(options, 'out');
+        }
+
+        if (outputType === 'array') {
+            return context.sendJson([{
+                label: 'Items Count',
+                value: 'count',
+                schema: { type: 'integer' }
+            }, {
+                label,
+                value: 'result',
+                schema: {
+                    type: 'array',
+                    items: {
+                        type: 'object',
+                        properties: itemSchema
+                    }
+                }
+            }], 'out');
+        }
+
+        if (outputType === 'file') {
+            return context.sendJson([{ label: 'File ID', value: 'fileId' }], 'out');
+        }
+    },
+
+    /**
+ * Normalize multiselect input (array or string) to array format for Canvas API.
+ * @param {string|string[]} input
+ * @param {number} maxItems
+ * @param {object} context
+ * @param {string} fieldName
+ * @returns {string[]}
+ */
+    normalizeMultiselectInput(input, maxItems, context, fieldName) {
+
+        let normalizedInput;
+
+        if (Array.isArray(input)) {
+            normalizedInput = input;
+        } else if (typeof input === 'string') {
+            // Handle comma-separated string
+            normalizedInput = input.split(',').map(item => item.trim()).filter(item => item.length > 0);
+        } else {
+            throw new context.CancelError(`${fieldName} must be a string or an array`);
+        }
+
+        if (maxItems !== Infinity && normalizedInput.length > maxItems) {
+            throw new context.CancelError(`Cannot have more than ${maxItems} items selected in ${fieldName}.`);
+        }
+
+        return normalizedInput;
+    }
+};
+
+/**
+ * @param {array} array
+ * @returns {string}
+ */
+const toCsv = (array) => {
+
+    const headers = Object.keys(array[0]);
```

Comment:
Add this fix from before: https://github.com/Appmixer-ai/appmixer-cli/pull/285/files#diff-4512df4d6ae77c395c6a99b3e9ab7a45ba7a0b5587e68fd76a4532e24f8ce504R110-R113